# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1315333839 14400
#      Tue Sep 06 14:30:39 2011 -0400
# Branch stable
# Node ID c7512d0d52e8cb862819e09415e7f6e990f6cab0
# Parent  8b7e6f359cee6727a4915e2931f63ba205f9b1a2
src/graphics.cc: Ensure tick separation is in factors of 10 for log plots.
Bug #34207.

diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -5118,16 +5118,23 @@ axes::properties::calc_ticks_and_lims (a
     {
       // FIXME we should check for negtives here
       hi = std::log10 (hi);
       lo = std::log10 (lo);
     }
 
   double tick_sep = calc_tick_sep (lo , hi);
 
+  if (is_logscale && ! (xisinf (hi) || xisinf (lo)))
+    {
+      // FIXME - what if (hi-lo) < tick_sep?
+      //         ex: loglog ([1 1.1])
+      tick_sep = std::ceil (tick_sep, 1.);
+    }
+
   int i1 = static_cast<int> (gnulib::floor (lo / tick_sep));
   int i2 = static_cast<int> (std::ceil (hi / tick_sep));
 
   if (limmode_is_auto)
     {
       // adjust limits to include min and max tics
       Matrix tmp_lims (1,2);
       tmp_lims(0) = std::min (tick_sep * i1, lo);
